### DISCUSSION TITLE
ExternalSecret is locked in degraded state

### DISCUSSION THREAD

**Discussion by mike-code** [argoproj/argo-cd#13785] [Q&A]
For some reason ExternalSecret resource gets deadlocked with state "secret already exists". I find it hard to reproduce as this issue happens c.a. 50% of times. 

I'm not sure why this happens but my wild guess was that ExternalSecret pulls a secret from external repository and  argocd treats it as "not mine" so it purges it and then some race condition happens?

I've also noticed in the logs following entries, note that they appear with 1 seconds difference, perhaps it is relevant.
```
time="2023-05-27T23:35:55Z" level=info msg="Applying resource ExternalSecret/apikey in cluster: https://10.186.32.1:443, namespace: app-881064370-e2e"
time="2023-05-27T23:35:56Z" level=info msg="Applying resource ExternalSecret/apikey in cluster: https://10.186.32.1:443, namespace: app-881064370-e2e"
```

![image](https://github.com/argoproj/argo-cd/assets/2964642/92d71dae-6023-4b73-8d41-149d75acfd6f)

**Comment by causalbody** [comment]
Seeing a similar issue

**Comment by m0un10** [comment]
This happens very often for us too. The only way I know how to fix it is to delete the secret from ArgoCD and then instantly it goes healthy (because it recreates the secrets fine).
